tabular/tests/unittests/metrics/test_confusion_matrix.py

```diff
@@ -0,0 +1,173 @@
+import pytest
+import numpy as np
+from autogluon.tabular import TabularPredictor, TabularDataset
+
+
+def test_confusion_matrix_basic(tmp_path):
+    """Test basic confusion matrix computation."""
+    train_data = TabularDataset({
+        'feature1': [1, 2, 3, 4, 5, 6, 7, 8],
+        'label':    [0, 1, 0, 1, 0, 1, 0, 1],
+    })
+
+    predictor = TabularPredictor(
+        label='label', 
+        path=tmp_path
+    ).fit(train_data, presets='medium_quality')
```

Comment:
I'll edit these tests later so it runs faster. No need to fit full medium quality, can fit a single dummy model instead.